File: Code9.Amazon.Web/Code9.Amazon.WebAPI/Helpers/VehicleParams.cs
namespace Code9.Amazon.WebAPI.Helpers
{
    public class VehicleParams
    {
        private const int MaxPageSize = 8;
        public int PageNumber { get; set; } = 1;
        private int pageSize = 8;
        public int PageSize
        {
            get { return pageSize; }
            set { pageSize = (value > MaxPageSize) ? MaxPageSize : value; }
        }

        public int Make { get; set; } = 0;
        public int Model { get; set; } = 0;
        public string FuelType { get; set; } = "";
        public string City { get; set; } = "";
        public int MaxMileage { get; set; } = 999999;
        public int MinPrice { get; set; } = 0;
        public int MaxPrice { get; set; } = 999999;
        public int MinYear { get; set; } = 1950;
        public int MaxYear { get; set; } = 2020;
        public string OrderBy { get; set; } = "";
    }
}

File: Code9.Amazon.Web/Code9.Amazon.WebSPA/Code9SPAProject/src/app/_models/vehicleParams.ts
export class VehicleParams {
  make: number;
  model: number;
  fuel: string;
  city: string;
  maxMileage: number;
  minPrice: number;
  maxPrice: number;
  minYear: number;
  maxYear: number;
  orderBy: string;

  constructor() {
    this.make = 0;
    this.model = 0;
    this.fuel = '';
    this.city = '';
    this.maxMileage = 999999;
    this.minPrice = 0;
    this.maxPrice = 999999;
    this.minYear = 1950;
    this.maxYear = 2020;
    this.orderBy = 'None';
  }
}
